### PR TITLE
chore(tests): add tests for ssr

### PR DESCRIPTION
**Pull request by tkrugg**
This adds some basic testing for SSR?
I basically tests two cases:
- SSR when accessing `/`
- SSR when accessing a specific page number `/?p=*`

### CODE REVIEW

File: examples/server-side-rendering/package.json [hunk 1]
```diff
@@ -55,12 +57,34 @@
...
+    },
+    "transformIgnorePatterns": [
+      "<rootDir>/lib/polyfills/*.js"
+    ],
+    "testRegex": "(/lib/__tests__/.*|(\\.|/)(test|spec))\\.(js?|ts?)$",
+    "moduleFileExtensions": [
+      "ts",
+      "js",
+      "json",
+      "node"
```
Comment (Haroenv) [inline examples/server-side-rendering/package.json]: why is `node` an extension? for jsdom?

File: examples/server-side-rendering/__tests__/ssr.spec.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import fetch from 'node-fetch';
+import { JSDOM } from 'jsdom';
+
+describe('SSR', () => {
+  describe('with no state', () => {
+    let dom;
+    beforeAll(async () => {
+      const res = await fetch('http://localhost:4000/');
+      const html = await res.text();
+      dom = new JSDOM(html);
```
Comment (Haroenv) [inline examples/server-side-rendering/__tests__/ssr.spec.ts]: could this have been done with https://yarnpkg.com/en/package/jest-environment-jsdom-thirteen and `document.innerHTML = ...`?

File: examples/server-side-rendering/package.json (hunk 1 above)
Comment (tkrugg) [reply examples/server-side-rendering/package.json]: I'm only using jsdom to parse the HTML into DOM objects easier to tests.
I don't actually need a browser context.

File: examples/server-side-rendering/__tests__/ssr.spec.ts (hunk 2 above)
Comment (tkrugg) [reply examples/server-side-rendering/__tests__/ssr.spec.ts]: same as responded previously, I don't think I need an actual browser context.
Is there a benefit I've missed into using a jsdom env?

Comment (Haroenv) [reply examples/server-side-rendering/__tests__/ssr.spec.ts]: seems like code could be simpler in the test

File: examples/server-side-rendering/package.json (hunk 1 above)
Comment (tkrugg) [reply examples/server-side-rendering/package.json]: sorry I didn't read you comment correctly. You're right.

**Pull request merged**